### ISSUE TITLE
stickyHeaderIndices breaks Transition inside FlatList for Android

### ISSUE THREAD

**Issue by sintylapse** [software-mansion/react-native-reanimated#806]
## Description
When using transitions api with `FlatList` with `stickyHeaderIndices` on Android,`Transition.Change` works bad when scroll offset is more than `0`. It seems like sticky header tries to animate to its original position and then goes back to its sticky position when any layout change is happened. 

### Screenshots
![Screen Recording 2020-05-25 at 18 16 54](https://user-images.githubusercontent.com/12295016/82842664-111b7b80-9ee3-11ea-937b-cc5116e0bdde.gif)

## Steps To Reproduce
Wrap `FlatList` with `Transitioning.View`, and apply `Transition.Change` animation to it. Add `stickyHeaderIndices` to `FlatList`, scroll a little bit, then add, or delete any item

### Expected behavior
Sticky component should not change its position

### Actual behavior
Sticky component just flying around when change is happened

## Snack or minimal code example
```js
transition = (
  <Transition.Together>
    <Transition.In type="fade" interpolation="easeInOut" durationMs={1000} />
    <Transition.Change interpolation="easeInOut" durationMs={1000} />
    <Transition.Out type="fade" interpolation="easeInOut" durationMs={1000} />
  </Transition.Together>
);

render() {
  return (
    <Transitioning.View ref={this.transitioning} transition={this.transition}>
      <FlatList
        data={records}
        renderItem={this.renderWishes}
        keyExtractor={this.generateKeyExtractor}
        stickyHeaderIndices={stickyHeaderIndices}
      />
    </Transitioning.View>
  )
}
```

## Package versions
- React: 16.9.0
- React Native: 0.61.5
- React Native Reanimated: 1.8.0

**Comment by sintylapse** [comment]
When using `LayoutAnimations` this buggy animation does not appear

**Comment by jakub-gonet** [comment]
We discussed Transitions internally in our team. Considering that v2 makes creating animations easier and Transitions was an experimental API for a long time, we decided to leave Transitions as they are.

We won’t actively fix bugs in them, but we’ll happily merge any PR with fixes. We’re also thinking about rewriting Transitions in v2 sometime in the future.

I’m closing this issue, but if you have further questions, I will be happy to help.
